src/Lib/Element.php

```diff
@@ -14,185 +14,150 @@
-        } elseif (is_string($serializable)) {
-            return $this->processElementsForString($serializable);
+        if ($isInternal === true) {
+            $this->elements[$elementName] = $element;
         } else {
-            throw new Exception\NotSupported('Cannot process elements for var of type "%s"', gettype($serializable));
+            $this->externalElements[$elementName] = $element;
         }
+
+        return $this;
     }
 
-    /**
-     * Add Elements to an array
-     *
-     * @param  array  $serializable
-     *
-     * @return array
-     */
-    private function processElementsForArray(array $serializable)
+    public function getExternalElements()
     {
-        foreach ($this->elements as $key => $element) {
-            $serializable[$key] = $element;
-        }
-        return $serializable;
+        return $this->externalElements;
     }
 
-    /**
-     * Add Elements to an Element
-     *
-     * @param  Element $serializable
-     *
-     * @return Element
-     */
-    private function processElementsForObject(Element $serializable)
+    public function getExternalOptions()
     {
-        foreach ($this->elements as $key => $element) {
-            $serializable->addElement($key, $element);
-        }
-        return $serializable;
+        return $this->externalOptions;
     }
 
-    /**
-     * Add Elements to a string
-     *
-     * @param  string $serializable
-     *
-     * @return mixed
-     */
-    private function processElementsForString($serializable)
+    public function hasOptions()
     {
-        if (count($this->elements) === 0) {
-            return $serializable;
-        } else {
-            return [
-                $serializable => $this->elements,
-            ];
-        }
+        return count($this->options) > 0;
     }
 
     /**
-     * Determine the type of serializable we have been passed, and append options to it accordingly
-     *
-     * @param  mixed $serializable
-     *
-     * @throws Exception\NotSupported
-     *
-     * @return mixed
+     * Extract any external elements from the elements and combine them with the elements array
+     * @return array
      */
-    private function processOptions($serializable)
+    protected function extractElements()
     {
-        if (is_array($serializable)) {
-            return $this->processOptionsForArray($serializable);
-        } elseif (is_object($serializable)) {
-            return $this->processOptionsForObject($serializable);
-        } elseif (is_string($serializable)) {
-            return $this->processOptionsForString($serializable);
-        } else {
-            throw new Exception\NotSupported('Cannot process options for var of type "%s"', gettype($serializable));
+        $elements = [];
+        foreach($this->elements as $element) {
+            $elements = array_merge($elements, $element->getExternalElements());
         }
+        $elements = array_merge($this->elements, $elements);
```

Comment:
I'm just confused on what is being merged here, we have $this->elements in the above foreach, but we merge it with the external elements first.